actioncable/lib/action_cable/test_helper.rb

```diff
@@ -0,0 +1,124 @@
+      #
+      #   def test_transmissions
+      #     assert_transmissions 0, 'messages'
+      #     ActionCable.server.broadcast 'messages', { text: 'hello' }
+      #     assert_transmissions 1, 'messages'
+      #     ActionCable.server.broadcast 'messages', { text: 'world' }
+      #     assert_transmissions 2, 'messages'
+      #   end
+      #
+      # If a block is passed, that block should cause the specified number of
+      # messages to be transmited.
+      #
+      #   def test_tramsmissions_again
+      #     assert_transmissions 1, 'messages'
+      #       ActionCable.server.broadcast 'messages', { text: 'hello' }
+      #     end
+      #
+      #     assert_transmissions 2, 'messages'
+      #       ActionCable.server.broadcast 'messages', { text: 'hi' }
+      #       ActionCable.server.broadcast 'messages', { text: 'how are you?' }
+      #     end
+      #   end
+      def assert_transmissions(number, channel)
+        if block_given?
+          original_count = transmissions_size(channel)
+          yield
+          new_count = transmissions_size(channel)
+          assert_equal number, new_count - original_count, "#{number} transmissions to #{channel} expected, but #{new_count - original_count} were sent"
+        else
+          actual_count = transmissions_size(channel)
+          assert_equal number, actual_count, "#{number} transmissions to #{channel} expected, but #{actual_count} were sent"
+        end
+      end
+
+      # Asserts that no messages have been sent to the channel.
+      #
+      #   def test_transmissions
+      #     assert_no_transmissions 'messages'
+      #     ActionCable.server.broadcast 'messages', { text: 'hi' }
+      #     assert_transmissions 1, 'messages'
+      #   end
+      #
+      # If a block is passed, that block should not cause any message to be sent.
+      #
+      #   def test_transmissions_again
+      #     assert_no_transmissions 'messages' do
+      #       # No job messages should be sent from this block
+      #     end
+      #   end
+      #
+      # Note: This assertion is simply a shortcut for:
+      #
+      #   assert_transmissions 0, 'messages', &block
+      def assert_no_transmissions(channel, &block)
+        assert_transmissions 0, channel, &block
+      end
+
+      # Asserts that the specified message has been sent to the channel.
+      #
+      #   def test_assert_transmited_message
+      #     ActionCable.server.broadcast 'messages', { text: 'hello' }
+      #     assert_transmited_data({text: 'hello'}, 'messages')
+      #   end
+      #
+      # If a block is passed, that block should cause a message with the specified data to be sent.
+      #
+      #   def test_assert_transmited_data_again
+      #     assert_transmited_data({text: 'hello'}, 'messages') do
+      #       ActionCable.server.broadcast 'messages', { text: 'hello' }
+      #     end
+      #   end
+      def assert_transmited_data(data, channel)
```

Comment:
Let's flip these parameters and go with `assert_transmission_on channel, data` so we don't have to wrap the hash.